### PR TITLE
legacy bot support

### PR DESCRIPTION
**Pull request by lexelby**
A change and bugfix to support our legacy chatops bot.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@ class User < ApplicationRecord
   validates :handle, format: { with: /\A[-_.a-z0-9]+\z/, message: "only allows numbers, lowercase letters, dashes, periods and underscores" }, allow_nil: true
 
   scope :with_handle, -> { where.not(handle: [nil, ""]) }
+  scope :with_name, -> { where.not(name: [nil, ""]) }
```
Comment (reidmix) [inline app/models/user.rb]: It'd be really nice to make all missing values `nil` then you don't have to test/scope for the empty string `""`. 
Can we normalize the `name`/`handle` when inserting into the db?

File: lib/chat_ops/chat_ops.rb [hunk 2]
```diff
@@ -15,11 +15,10 @@ def commands
...
 
     # Generates a regex that matches any of the declared ChatOps::Command
-    # subclassess.  Each ChatOps::Command subclass has its own regex, so we use
-    # Regexp.union to combine them.  It's almost the same as joining them with
-    # '|'.
+    # subclassess.  We massage the regex a bit so that it's more palatable to
+    # regex implementations with fewer features than Ruby's.
     def matcher
-      Regexp.union(commands.collect(&:regex))
+      "(?ix)" + commands.map(&:regex).map(&:source).join('|').gsub(/\(\?<[^>]+>/, '(')
```
Comment (reidmix) [inline lib/chat_ops/chat_ops.rb]: can you help me understand why we are getting rid of named group `(?<thing>` parts of regex?
I guess i don't understand the intent.

Comment (reidmix) [inline lib/chat_ops/chat_ops.rb]: Also, could you help me understand why we have to make it "more palatable to regex implementations with fewer features than Ruby's"?  I'm guessing these are related :D

Comment (reidmix) [inline lib/chat_ops/chat_ops.rb]: Oh, I think I get it, `matcher` is for use outside Retrodot (presumably heimdall), and the named groups are making it puke.  Is that right?  If so, can we put a comment that matcher is intended for external use?

File: app/models/user.rb (hunk 1 above)
Comment (lexelby) [inline app/models/user.rb]: I'll look into this.

File: lib/chat_ops/chat_ops.rb (hunk 2 above)
Comment (lexelby) [inline lib/chat_ops/chat_ops.rb]: Yup, you got it.  I'll clarify the comment.

**Comment by joshuatobin** [comment]
👍
